internal_packages/keybase/lib/preferences-keybase.cjsx

```diff
@@ -0,0 +1,238 @@
+      VsVlQKtXimFLpck1z0BPQ0NmRdEzRHQwP2WNYfxdNCeFAGDL4tpblBzw/vp/CFTO
+      217s2OKjpJqtpHPf2vY=
+      =UY7Y
+      -----END PGP PRIVATE KEY BLOCK-----"""
+
+  constructor: (@props) ->
+    @_keySaveQueue = {}
+
+    {pubKeys, privKeys, selectedPubKey, selectedPrivKey} = @_getStateFromStores()
+    @state =
+      pubKeys: pubKeys
+      privKeys: privKeys
+
+      selectedPubKey: selectedPubKey
+      selectedPrivKey: selectedPrivKey
+
+  componentWillMount: =>
+    @unlistenKeystore = PGPKeyStore.listen(@_onChange, @)
+
+  componentWillUnmount: =>
+    @unlistenKeystore()
+
+  _onChange: =>
+    @setState @_getStateFromStores()
+
+  _getStateFromStores: ->
+    pubKeys = PGPKeyStore.pubKeys()
+    privKeys = PGPKeyStore.privKeys(timed: false)
+
+    _.each(pubKeys, (key) =>
+      if not key.key?
+        PGPKeyStore.getKeyContents(key: key)
+    )
+
+    selectedPubKey = @state?.selectedPubKey
+    # deleted
+    ###
+    if selectedPubKey? and selectedPubKey.address not in _.pluck(pubKeys, "address")
+      selectedPubKey = null
+    ###
+    # none selected
+    if not selectedPubKey
+      selectedPubKey = if pubKeys.length > 0 then pubKeys[0]
+
+    selectedPrivKey = @state?.selectedPrivKey
+    ###
+    if selectedPrivKey? and selectedPrivKey.address not in _.pluck(privKeys, "address")
+      selectedPrivKey = null
+    ###
+    if not selectedPrivKey
+      selectedPrivKey = if privKeys.length > 0 then privKeys[0] else null
+
+    return {pubKeys, privKeys, selectedPubKey, selectedPrivKey}
+
+  componentDidMount: ->
+    @usub = PGPKeyStore.listen @_onChange
+
+  # Key Editing
+  _onSelectPubKey: (event) =>
+    selectedPubKey = null
+    for pubKey in @state.pubKeys
+      if event.target.value in privKey.addresses
+        selectedPubKey = pubKey
+    @setState
+      selectedPubKey: selectedPubKey
+
+  _onSelectPrivKey: (event) =>
+    selectedPrivKey = null
+    for privKey in @state.privKeys
+      if event.target.value in privKey.addresses
+        selectedPrivKey = privKey
+    @setState
+      selectedPrivKey: selectedPrivKey
+
+  # DELETE AND NEW
+  _deletePubKey: =>
+    numKeys = @state.pubKeys.length
+    if @state.selectedPubKey?
+      PGPKeyStore.deleteKey(@state.selectedPubKey)
+    if numKeys==1
+      @setState
+        selectedPubKey: null
+
+  _deletePrivKey: =>
+    numKeys = @state.privKeys.length
+    if @state.selectedPrivKey?
+      PGPKeyStore.deleteKey(@state.selectedPrivKey)
+    if numKeys==1
+      @setState
+        selectedPrivKey: null
+
+  _saveNewPubKey: =>
+    PGPKeyStore.saveNewKey("benbitdiddle@icloud.com", TEST_KEY, isPub = true)
+
+  _saveNewPrivKey: =>
+    PGPKeyStore.saveNewKey("benbitdiddle@icloud.com", TEST_KEY, isPub = false)
+
+  _addOtherAddress: =>
+    PGPKeyStore.addAddressToKey(@state.pubKeys[0], "logan@nylas.com")
+
+  _removeOtherAddress: =>
+    PGPKeyStore.removeAddressFromKey(@state.pubKeys[0], "logan@nylas.com")
```

Comment:
TODO add `ParticipantsTextField` to the key add/remove/etc. components Logan set up and remove these hardcoded values.